lib/lltz_michelson/lltz_michelson.ml

```diff
@@ -181,4 +181,80 @@ let convert_primitive (prim: LLTZ.P.t) : Michelson.Ast.t =
     | Get_and_update -> get_and_update
     | Transfer_tokens -> transfer_tokens
     | Check_signature -> check_signature
-    | Open_chest -> open_chest
\ No newline at end of file
+    | Open_chest -> open_chest
+
+let rec compile : LLTZ.E.t -> Instruction.t =
+  fun expr ->
+   let open Instruction in
+   Instruction.seq
+     [ 
+       match expr.desc with 
+       | Variable (Var name) -> Instruction.Slot.dup (`Ident name)
+       | Let_in { let_var = Var var; rhs; in_ } ->
+         Instruction.seq [ 
+          compile rhs;  
+          Instruction.Slot.let_ (`Ident var) 
+          ~in_:( compile in_ ) 
+          ]
+       | Lambda { lam_var = (Var var, lam_var_type); return_type; body } -> 
+         assert false
+       | Lambda_rec { lam_var = (Var var, lam_var_type); mu_var = Var mu; return_type; body } -> 
+         assert false
+       | App { abs; arg } ->
+          assert false
+       | Const constant -> 
+         seq [ 
+           push (get_const_type constant) (convert_constant constant) 
+          ]
+       | Prim (primitive, args) ->
+         let args_instrs = List.map args ~f:compile in
+         seq (
+           args_instrs @ 
+           [ prim (List.length args) 1 (convert_primitive primitive) ])
+       | Let_mut_in { let_var = Mut_var var; rhs; in_ } ->
+        assert false
+       | Deref (Mut_var var) -> Slot.dup (`Ident var)
```

Comment:
Please, explain each case. This one is very short but needs an explanation nonetheless.